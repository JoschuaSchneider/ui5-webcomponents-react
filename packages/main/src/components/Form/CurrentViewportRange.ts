import React from 'react';

const CurrentRange = React.createContext(null);

export default CurrentRange;
