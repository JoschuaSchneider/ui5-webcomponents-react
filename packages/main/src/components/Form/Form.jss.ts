export const styles = {
  formTitle: {
    borderBottom: '1px solid grey'
  },
  formTitlePaddingBottom: {
    paddingBottom: '2em'
  },
  formPaddingBottom: {
    paddingBottom: '1em'
  },
  formGroupStyle: {
    width: '100%',
    paddingTop: '0.25em'
  },
  formItemTopDiv: {
    alignItems: 'center'
  },
  formLabel: {
    paddingRight: '0.5em'
  },
  formElement: {
    display: 'block'
  }
};
