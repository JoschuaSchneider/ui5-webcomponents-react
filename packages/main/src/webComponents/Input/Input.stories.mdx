import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import { InputType } from '@ui5/webcomponents-react/lib/InputType';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import '@ui5/webcomponents-icons/dist/icons/employee.js';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import { SuggestionItem } from '@ui5/webcomponents-react/lib/SuggestionItem';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';

<Meta
  title="UI5 Web Components / Input"
  component={Input}
  subcomponents={{ SuggestionItem }}
  argTypes={{
    ...createSelectArgTypes({ type: InputType, valueState: ValueState }),
    children: { control: { disable: true } },
    icon: { control: { disable: true } },
    valueStateMessage: { control: { disable: true } }
  }}
  args={{
    type: InputType.Text,
    valueState: ValueState.None,
    icon: <Icon name="employee" />
  }}
/>

<DocsHeader />

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Input {...args} />;
    }}
  </Story>
</Canvas>

<ArgsTable story="." />
