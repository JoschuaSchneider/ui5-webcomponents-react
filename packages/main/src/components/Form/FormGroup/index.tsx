import React, { Children, FC, forwardRef, ReactNode, ReactNodeArray, Ref } from 'react';
import { FlexBox, FlexBoxAlignItems, FlexBoxDirection, FlexBoxJustifyContent, Title, TitleLevel } from '../../..';
import { styles } from '../Form.jss';

export interface FormGroupProps {
  id?: string;
  title?: string;
  children: ReactNode | ReactNodeArray;
  type?: string;
}

const FormGroup: FC<FormGroupProps> = forwardRef((props: FormGroupProps, ref: Ref<HTMLDivElement>) => {
  const { id, title, children } = props;

  return (
    <div id={id} ref={ref}>
      {title ? (
        <Title level={TitleLevel.H5} style={styles.formPaddingBottom}>
          {title}
        </Title>
      ) : null}
      <FlexBox
        justifyContent={FlexBoxJustifyContent.Start}
        alignItems={FlexBoxAlignItems.End}
        fitContainer={true}
        direction={FlexBoxDirection.Column}
      >
        {Children.map(children, (child) => {
          return <div style={styles.formGroupStyle}>{child}</div>;
        })}
      </FlexBox>
    </div>
  );
});

FormGroup.defaultProps = {
  type: 'formGroup'
};

export { FormGroup };
