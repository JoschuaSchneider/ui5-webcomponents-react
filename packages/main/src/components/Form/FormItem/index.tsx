import React, {
  Children,
  FC,
  forwardRef,
  Ref,
  ReactElement,
  ReactNode,
  useState,
  CSSProperties,
  useCallback,
  ReactNodeArray
} from 'react';
import { useRateChanged } from '../hooks/useRateChanged';
import { Label } from '../../..';
import { styles } from '../Form.jss';

export interface FormItemProps {
  id?: string;
  labelText?: string;
  children: ReactNode | ReactNodeArray;
  type?: string;
}

const calculateWidth = (rate) => {
  return Math.floor((100 / 12) * rate) + '%';
};

const FormItem: FC<FormItemProps> = forwardRef((props: FormItemProps, ref: Ref<HTMLDivElement>) => {
  const { id, labelText, children } = props;

  const [currentRate, setCurrentRate] = useState('');
  const subscribe = useRateChanged()[1];
  const onRateChange = useCallback(
    (rate) => {
      setCurrentRate(rate);
    },
    [currentRate]
  );

  const [initSubscribe, setInitSubscribe] = useState(true);
  if (initSubscribe) {
    subscribe(onRateChange);
    setInitSubscribe(false);
  }

  const renderChildren = (child: ReactElement<any>) => {
    if (currentRate === '') return '';
    let labelWidth,
      labelTextAlign = 'flex-end',
      display = 'flex',
      elementWidth;

    switch (currentRate) {
      case 'S':
        labelWidth = '100%';
        elementWidth = '100%';
        display = 'block';
        labelTextAlign = 'flex-start';
        break;
      case 'M':
        labelWidth = calculateWidth(2);
        elementWidth = calculateWidth(10);
        break;
      case 'L':
      case 'XL':
        labelWidth = calculateWidth(4);
        elementWidth = calculateWidth(8);
        break;
    }

    const topDivStyle = { ...styles.formItemTopDiv, display: display } as CSSProperties;
    const labelStyle = { ...styles.formLabel, width: labelWidth, justifyContent: labelTextAlign } as CSSProperties;
    const elementStyle = { ...styles.formElement, width: elementWidth } as CSSProperties;

    return (
      <div style={topDivStyle}>
        <Label style={labelStyle}>{labelText ? labelText : ''}</Label>
        <div style={elementStyle}>{child}</div>
      </div>
    );
  };

  return (
    <div id={id} ref={ref}>
      {Children.map(children, renderChildren)}
    </div>
  );
});

FormItem.defaultProps = {
  type: 'formItem'
};

export { FormItem };
