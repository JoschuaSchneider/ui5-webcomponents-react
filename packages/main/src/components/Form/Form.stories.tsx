import { Form } from './index';
import React from 'react';
import notes from './Form.md';
import { FormItem } from './FormItem';
import { FormGroup } from './FormGroup';
import { CheckBox, Input, InputType, Select, Option } from '../..';

export const defaultStory = () => {
  return (
    <Form title={'Test Form'}>
      <FormGroup title={'Personal Data'}>
        <FormItem labelText={'Name'} id={'name'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Address'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Country'}>
          <Select>
            <Option>Germany</Option>
            <Option>France</Option>
            <Option>Italy</Option>
          </Select>
        </FormItem>
        <FormItem labelText={'Home address'}>
          <CheckBox checked={true} />
        </FormItem>
      </FormGroup>
      <FormGroup title={'Business Data'}>
        <FormItem labelText={'Organization'} id={'name'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Position'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Wage'}>
          <Input type={InputType.Number} value={'5000'} disabled={true} />
        </FormItem>
        <FormItem labelText={'Pilot license'}>
          <CheckBox checked={true} />
        </FormItem>
      </FormGroup>
    </Form>
  );
};

export default {
  title: 'Components | Form',
  component: Form,
  parameters: { notes }
};
