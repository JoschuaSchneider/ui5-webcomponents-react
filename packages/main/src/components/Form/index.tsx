import { Grid } from '../Grid';
import React, { FC, forwardRef, ReactElement, ReactNode, ReactNodeArray, Ref } from 'react';
import { CommonProps } from '../../interfaces/CommonProps';
import { Title, TitleLevel } from '../..';
import { useRateChanged } from './hooks/useRateChanged';
import { styles } from './Form.jss';
import { FormGroup } from './FormGroup';
import { FormItem } from './FormItem';

export interface FormPropTypes extends CommonProps {
  /**
   * Components that are placed into Form.
   */
  children: ReactNode | ReactNodeArray;
  /**
   * Form title
   */
  title?: string;
}

const Form: FC<FormPropTypes> = forwardRef((props: FormPropTypes, ref: Ref<HTMLDivElement>) => {
  const { title, children } = props;

  const rateChanged = useRateChanged()[0];
  const onRateChanged = (e) => {
    rateChanged(e.parameters.rate);
  };

  let ungroupedChildren = [],
    updatedChildren,
    newTitle;
  if (children.hasOwnProperty('length')) {
    updatedChildren = [...children];
    for (let i = updatedChildren.length - 1; i >= 0; i--) {
      if (updatedChildren[i].props.type === 'formItem') {
        ungroupedChildren.push(updatedChildren.splice(i, 1)[0]);
      }
    }
    if (ungroupedChildren.length > 0) {
      updatedChildren.push(<FormGroup children={ungroupedChildren.reverse()} />);
    }
    newTitle = title;
  } else {
    if ((!title || title.length === 0) && children.props.title && children.props.title.length > 0) {
      newTitle = children.props.title;
      updatedChildren = React.cloneElement(children, { title: null });
    } else newTitle = title;
  }

  return (
    <React.Fragment>
      {newTitle ? (
        <React.Fragment>
          <Title level={TitleLevel.H3} style={styles.formTitle}>
            {newTitle}
          </Title>{' '}
          <div style={styles.formTitlePaddingBottom} />
        </React.Fragment>
      ) : null}
      <Grid
        ref={ref}
        children={updatedChildren ? updatedChildren : children}
        defaultSpan={'XL6 L12 M12 S12'}
        onRateChanged={onRateChanged}
      />
    </React.Fragment>
  );
});

Form.defaultProps = {
  children: [],
  title: null
};
Form.displayName = 'Form';

export { Form, FormGroup, FormItem };
