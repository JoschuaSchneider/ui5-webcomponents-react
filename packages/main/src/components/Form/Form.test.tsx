import { mountThemedComponent } from '@shared/tests/utils';
import * as React from 'react';
//import {Form, FormGroup, FormItem} from '@ui5/webcomponents-react/lib/Form';
import { Form, FormGroup, FormItem } from '../Form';
import { Input } from '../../webComponents/Input';
import { InputType } from '../..';

const SIZE_S = 200;
const SIZE_M = 800;
const SIZE_L = 1200;
const SIZE_XL = 1600;
const component = (
  <Form title={'Test form'}>
    <FormGroup title={'Group 1'}>
      <FormItem labelText={'item 1'}>
        <Input type={InputType.Text}></Input>
      </FormItem>
      <FormItem labelText={'item 2'}>
        <Input type={InputType.Number}></Input>
      </FormItem>
    </FormGroup>
    <FormGroup title={'Group 2'}>
      <FormItem labelText={'item 1'}>
        <Input type={InputType.Text}></Input>
      </FormItem>
      <FormItem labelText={'item 2'}>
        <Input type={InputType.Number}></Input>
      </FormItem>
    </FormGroup>
  </Form>
);

describe('Create a Form', () => {
  test('size rate S; should create Label and Element with 100% width and display: block for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_S });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('size rate M; should create Label and Element with 16% and 83% width respectively and display: flex for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_M });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('size rate L; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_L });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('size rate XL; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_XL });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
