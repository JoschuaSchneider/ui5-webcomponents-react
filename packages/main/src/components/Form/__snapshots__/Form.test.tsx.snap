// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a Form size rate L; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div 1`] = `
Array [
  <ui5-title
    class=""
    level="H3"
    style="border-bottom: 1px solid grey;"
  >
    Test form
  </ui5-title>,
  <div
    style="padding-bottom: 2em;"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form size rate M; should create Label and Element with 16% and 83% width respectively and display: flex for top FormItem div 1`] = `
Array [
  <ui5-title
    class=""
    level="H3"
    style="border-bottom: 1px solid grey;"
  >
    Test form
  </ui5-title>,
  <div
    style="padding-bottom: 2em;"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 16%; text-align: end; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 83%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 16%; text-align: end; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 83%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 16%; text-align: end; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 83%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 16%; text-align: end; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 83%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form size rate S; should create Label and Element with 100% width and display: block for top FormItem div 1`] = `
Array [
  <ui5-title
    class=""
    level="H3"
    style="border-bottom: 1px solid grey;"
  >
    Test form
  </ui5-title>,
  <div
    style="padding-bottom: 2em;"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: block; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 100%; text-align: start; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 100%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: block; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 100%; text-align: start; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 100%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: block; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 100%; text-align: start; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 100%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: block; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 100%; text-align: start; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 100%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form size rate XL; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div 1`] = `
Array [
  <ui5-title
    class=""
    level="H3"
    style="border-bottom: 1px solid grey;"
  >
    Test form
  </ui5-title>,
  <div
    style="padding-bottom: 2em;"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class=""
          level="H5"
          style="padding-bottom: 1em;"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 1
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Text"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 100%; padding-top: 0.25em;"
          >
            <div>
              <div
                style="display: flex; align-items: center;"
              >
                <ui5-label
                  class=""
                  for=""
                  style="width: 33%; text-align: end; padding-right: 0.4em;"
                >
                  item 2
                </ui5-label>
                <div
                  style="width: 66%; display: block;"
                >
                  <ui5-input
                    class=""
                    type="Number"
                    value=""
                    value-state="None"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
